### PR TITLE
Add listeners to prevent un-authorized update/delete of system applications

### PR DESCRIPTION
**Pull request by thanujalk**

### CODE REVIEW

**Review by DimalChandrasiri — APPROVED** [review]

File: components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/listner/AppPortalApplicationMgtListener.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    public AppPortalApplicationMgtListener(boolean enable) {
+
+        this.enable = enable;
+        systemApplications = AppsCommonDataHolder.getInstance().getSystemApplications();
+    }
+
+    @Override
+    public int getDefaultOrderId() {
+        return 30;
```
Comment (pamodaaw) [inline components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/listner/AppPortalApplicationMgtListener.java:48]: keep new line after method declaration

File: components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/listner/AppPortalOAuthAppMgtListener.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            throws IdentityOAuthAdminException {
+
+        if (!isEnabled() || systemAppConsumerKeys.stream()
+                .noneMatch(oAuthConsumerAppDTO.getApplicationName()::equalsIgnoreCase)) {
+            return;
+        }
+
+        throw new IdentityOAuthClientException(Error.INVALID_UPDATE.getErrorCode(),
+                "System application update is not allowed. Client id: " + oAuthConsumerAppDTO
+                        .getOauthConsumerKey());
```
Comment (pamodaaw) [inline components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/listner/AppPortalOAuthAppMgtListener.java:68]: formatting issue

File: components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/util/AppPortalUtils.java [hunk 3]
```diff
@@ -226,4 +227,31 @@ public static void initiatePortals(String tenantDomain, int tenantId)
...
+     * @param application application.
+     * @return OAuth InboundAuthenticationRequestConfig if exists.
+     */
+    public static InboundAuthenticationRequestConfig getOAuthInboundAuthenticationRequestConfig(
+            ServiceProvider application) {
+
+        if (application == null || application.getInboundAuthenticationConfig() == null
+                || application.getInboundAuthenticationConfig().getInboundAuthenticationRequestConfigs() == null
+                || application.getInboundAuthenticationConfig().getInboundAuthenticationRequestConfigs().length == 0) {
+
```
Comment (pamodaaw) [inline components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/util/AppPortalUtils.java:243]: do we need these new lines before the return?

File: components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/listner/AppPortalApplicationMgtListener.java (hunk 1 above)
Comment (thanujalk) [reply components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/listner/AppPortalApplicationMgtListener.java:49]: Done

File: components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/util/AppPortalUtils.java (hunk 3 above)
Comment (thanujalk) [reply components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/util/AppPortalUtils.java:243]: much readable that way.

**Review by pamodaaw — APPROVED** [review]

**Comment by jeradrutnam** [comment]
Related: https://github.com/wso2/product-is/issues/8716

**Pull request merged**